core/src/main/java/com/datastax/oss/driver/internal/core/ContactPoints.java

```
@@ -41,7 +38,22 @@ public static Set<EndPoint> merge(
 
     Set<EndPoint> result = Sets.newHashSet(programmaticContactPoints);
     for (String spec : configContactPoints) {
-      for (InetSocketAddress address : extract(spec, resolve)) {
+
+      Set<InetSocketAddress> addresses = Collections.emptySet();
+      try {
+        addresses = AddressUtils.extract(spec, resolve);
+      } catch (RuntimeException e) {
+        LOG.warn("Ignoring invalid contact point {} ({})", spec, e.getMessage(), e);
+      }
```

Could just continue here to next iteration of the outer for loop.  You know addresses is the empty set at this point so there's no point iterating over it below.

---

As I look at this now it feels like we had to make this a bit more complicated because AddressUtils.extract() now throws exceptions in most cases rather than just logging errors and returning an empty set.  Was there a particular reason for this change?  It's not immediately clear the exceptions buy you much here.

---

Previously, `#extract` code was used only in this class and we logged errors together with reasons of these errors. Now the info about reasons is moved to until method, which is called from multiple places. In this class, I aimed to keep logging (as well as other functionality) as close to the origin as seemed possible to avoid opinionated refactoring, so I needed a way to get reasons of errors from the utility `#extract` to log them together with the context logs.
Happy to agree on the way it should look like and change accordingly.